BBC.js

```diff
@@ -46,20 +46,37 @@ function detectWeb(doc, url) {
 		}
 		return "newspaperArticle";
 	}
-	if(url.indexOf("/newsbeat/article") != -1)
+	if(url.indexOf("/newsbeat/article") != -1){
 		return "blogPost";
+	}
 	if (getSearchResults(doc, true)) {
 		return "multiple";
 	}
 }
 
 function getSearchResults(doc, checkOnly) {
+	var namespace = doc.documentElement.namespaceURI;
+	var nsResolver = namespace ? function(prefix) {
+	if (prefix == 'x') return namespace; else return null;
+	} : null;
+
 	var items = {};
 	var found = false;
 	var rows = ZU.xpath(doc, '//a[h3[@class="title-link__title"]]');
-	for (var i=0; i<rows.length; i++) {
+	//for NewsBeat
```

Comment:
thanks, sorry, hadn't realized that test was a newsbeat search. I understand why you used the code; I just want to see if we can make this more robust.